tests/test_psf_trampoline.py

```diff
@@ -0,0 +1,141 @@
+# This file is part of afw.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import unittest
+
+import numpy as np
+
+import lsst.utils.tests
+from lsst.afw.detection import Psf, GaussianPsf
+from lsst.afw.image import Image
+from lsst.geom import Box2I, Point2I, Extent2I
+from lsst.afw.geom.ellipses import Quadrupole
+
+
+# Subclass Psf in python.  Main tests here are that python virtual methods get
+# resolved by trampoline class.
+class PyGaussianPsf(Psf):
+    def __init__(self, width, height, sigma):
+        Psf.__init__(self)
+        self.dimensions = Extent2I(width, height)
+        self.sigma = sigma
+
+    # "public" virtual overrides
+    def clone(self):
+        return PyGaussianPsf(self.dimensions.x, self.dimensions.y, self.sigma)
+
+    def resized(self, width, height):
+        return PyGaussianPsf(width, height, self.sigma)
+
+    # "private" virtual overrides are underscored
+    def _doComputeKernelImage(self, position=None, color=None):
+        bbox = self.computeBBox()
+        img = Image(bbox, dtype=np.float64)
+        x, y = np.ogrid[bbox.minY:bbox.maxY+1, bbox.minX:bbox.maxX+1]
+        rsqr = x**2 + y**2
+        img.array[:] = np.exp(-0.5*rsqr/self.sigma**2)
+        img.array /= np.sum(img.array)
+        return img
+
+    def _doComputeBBox(self, position=None, color=None):
+        return Box2I(Point2I(-self.dimensions/2), self.dimensions)
+
+    def _doComputeShape(self, position=None, color=None):
+        return Quadrupole(self.sigma**2, self.sigma**2, 0.0)
+
+    def _doComputeApertureFlux(self, radius, position=None, color=None):
+        return 1 - np.exp(-0.5*(radius/self.sigma)**2)
+
+
+class PythonPsfInheritTestSuite(lsst.utils.tests.TestCase):
+    def testTrampoline(self):
+        params = [
+            (5, 5, 1.1),
+            (5, 3, 1.2),
+            (7, 7, 1.3)
+        ]
+        for width, height, sigma in params:
+            pgp = PyGaussianPsf(width, height, sigma)
+            gp = GaussianPsf(width, height, sigma)
```

Comment:
Please consider factoring this out into a `setUp` method, then implementing each test case as an individual method. It will make it much easier to add new independent tests, and to diagnose individual failures.